usagov/usagov-2021
# USAGOV-2050-Cypress-Github-Actions: Github Actions Files

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-2050

## Description

- Add github action workflow files for regression testing stage and production static sites (https://beta-stage.usa.gov/ & https://usa.gov/). 
- The reason there are two extremely similar files "stage-a11y-regression-tests.yml & prod-a11y-regression-tests.yml" is because we change the CYPRESS_BASE_URL environment variable between the two. We aren't actually running against the code we are running against the static sites so we can't make use of the environment flag github actions has.
- If we had only one workflow file with two different jobs for the environments we would need to run both jobs each time. We don't want to run prod on a stage regression test day and vice-versa so two workflows are needed. 

## Type of Changes

  - Add screenshot showing what it should look like

## Testing Instructions



### Change Requirements



### Validation Steps

- This can't really be tested until deployed to the main branch (that's when the Github Action will appear) but for an idea of what this will do see [this repo ](https://github.com/IsabelLaurenceau/cypress-usagov/actions) where I worked on this. ( I didn't want to spam the team while doing this work so I made a separate repo to only spam myself) 
- You should be able to go to the actions tab and select either "Stage A11Y Regression Tests" or "Prod A11Y Regression Tests" from the left menu. 
- A "Run workflow" button will be on that page. If you click it and refresh the page you will see the regression test start. It should take about 3min and 30seconds. 
- I've noticed we have a few new failing tests but these actually need the tests to be updated not the Github action which will be done in a new ticket. 



## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
